pandas/core/arrays/interval.py

```diff
@@ -223,12 +229,14 @@ def _simple_new(
         if dtype is not None:
             # GH 19262: dtype must be an IntervalDtype to override inferred
             dtype = pandas_dtype(dtype)
-            if not is_interval_dtype(dtype):
+            if is_interval_dtype(dtype):
```

Comment:
could just do `isinstance(dtype, IntervalDtype)` and avoid cast

Reply:
In #38680, we discussed the usage of isinstance vs is_* methods in combination with cast and the latter was preferred. Therefore I continued with it here.